tests/config/test_omegaconf_config.py

```diff
@@ -649,3 +649,40 @@ def test_variable_interpolation_in_catalog_with_separate_templates_file(
         conf = OmegaConfigLoader(str(tmp_path))
         conf.default_run_env = ""
         assert conf["catalog"]["companies"]["type"] == "pandas.CSVDataSet"
+
+    def test_custom_resolvers(self, tmp_path):
+        base_params = tmp_path / _BASE_ENV / "parameters.yml"
+        param_config = {
+            "model_options": {
+                "test_size": "${add: 3, 4}",
+                "random_state": "${plus_2: 1}",
+            }
+        }
+        _write_yaml(base_params, param_config)
+        custom_resolvers = {
+            "add": lambda *x: sum(x),
+            "plus_2": lambda x: x + 2,
+        }
+        conf = OmegaConfigLoader(str(tmp_path), custom_resolvers=custom_resolvers)
+        conf.default_run_env = ""
+        assert conf["parameters"]["model_options"]["test_size"] == 7
+        assert conf["parameters"]["model_options"]["random_state"] == 3
+
+    def test_overwrite_resolvers(self, tmp_path):
+        base_params = tmp_path / _BASE_ENV / "parameters.yml"
+        # OmegaConf is a singleton, register a resolver to be overwritten
+        OmegaConf.register_new_resolver("custom", lambda x: x + 10)
+
+        param_config = {
+            "model_options": {
+                "test_size": "${custom: 10}",
+            }
+        }
+        _write_yaml(base_params, param_config)
+        custom_resolvers = {
+            "custom": lambda x: x + 20,
+        }
+        conf = OmegaConfigLoader(str(tmp_path), custom_resolvers=custom_resolvers)
+        conf.default_run_env = ""
+        # test_size should be calculated using overwritten custom resolver (x + 20)
+        assert conf["parameters"]["model_options"]["test_size"] == 30
```

Comment:
Is it possible to add a check/assert first to show that "test_size" is set to 20 and then after the overwriting it will be 30? Maybe just by calling omegaconf directly on the `param_config`?

Reply:
Updated the test